### PR TITLE
DROOLS-1692: Migrate tests in drools-persistence-jpa module from Bitr…

### PR DESCRIPTION
**Pull request by domhanak**
…onix to Narayana

@baldimir Please review, I added what was missing on top of your commits and squashed it. Tested locally with mariadb. Do you want me to add profile for local testing with docker mariadb into pom.xml to provide this option to someone else? 
I didn't apply any code-formatting so tell me if I should. This should go to 7.3.x aswell, I'll create a PRQ there once this gets approved.

Depends on https://github.com/kiegroup/optaplanner/pull/343

### CODE REVIEW

File: drools-persistence/drools-persistence-jpa/src/test/java/org/drools/persistence/marshalling/util/MarshallingDBUtil.java [hunk 1]
```diff
@@ -16,11 +16,12 @@
 
 package org.drools.persistence.marshalling.util;
 
-import bitronix.tm.resource.jdbc.PoolingDataSource;
+import static org.drools.persistence.util.DroolsPersistenceUtil.DATASOURCE;
```
Comment (baldimir) [inline drools-persistence/drools-persistence-jpa/src/test/java/org/drools/persistence/marshalling/util/MarshallingDBUtil.java]: Please revert this file, it's just imports change.

**Review by baldimir — CHANGES_REQUESTED** [review]
Please see my comment, otherwise looks good, thanks!

**Review by baldimir — APPROVED** [review]
Looks good now, +1.

**Comment by domhanak** [comment]
@mariofusco Could you take a look please and merge if you have no objections?

**Comment by mariofusco** [comment]
jenkins execute full downstream build

**Comment by mariofusco** [comment]
This pull request looks ok, but it is causing the following error in optaplanner, please fix it.

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.0:testCompile (default-testCompile) on project optaplanner-persistence-jpa: Compilation failure
11:03:50 [ERROR] /home/jenkins/workspace/drools-downstream-pullrequests/downstream-repos/optaplanner/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateTypeTest.java:[51,78] cannot find symbol
11:03:50 [ERROR] symbol:   variable TRANSACTION_MANAGER
11:03:50 [ERROR] location: class org.kie.test.util.db.PersistenceUtil
11:03:50 [ERROR] -> [Help 1]
11:03:50 org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.0:testCompile (default-testCompile) on project optaplanner-persistence-jpa: Compilation failure
11:03:50 /home/jenkins/workspace/drools-downstream-pullrequests/downstream-repos/optaplanner/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateTypeTest.java:[51,78] cannot find symbol
11:03:50   symbol:   variable TRANSACTION_MANAGER
11:03:50   location: class org.kie.test.util.db.PersistenceUtil
11:03:50 
11:03:50 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
11:03:50 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
11:03:50 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
11:03:50 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
11:03:50 	at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:185)
11:03:50 	at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:181)
11:03:50 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
11:03:50 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
11:03:50 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
11:03:50 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
11:03:50 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
11:03:50 	at java.lang.Thread.run(Thread.java:748)
11:03:50 Caused by: org.apache.maven.plugin.compiler.CompilationFailureException: Compilation failure
11:03:50 /home/jenkins/workspace/drools-downstream-pullrequests/downstream-repos/optaplanner/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateTypeTest.java:[51,78] cannot find symbol
11:03:50   symbol:   variable TRANSACTION_MANAGER
11:03:50   location: class org.kie.test.util.db.PersistenceUtil
11:03:50 
11:03:50 	at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:1029)
11:03:50 	at org.apache.maven.plugin.compiler.TestCompilerMojo.execute(TestCompilerMojo.java:164)
11:03:50 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
11:03:50 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
11:03:50 	... 11 more
```

**Comment by domhanak** [comment]
@mariofusco Thanks, I'll take a look.

**Comment by domhanak** [comment]
jenkins execute full downstream build

**Comment by domhanak** [comment]
Looks like there are some failures in optaplanner, I will check how it can be fixed

**Comment by domhanak** [comment]
jenkins execute full downstream build

**Comment by domhanak** [comment]
Not sure why downstream build failed, but the optaplanner tests are fixed.

**Comment by domhanak** [comment]
jenkins execute full downstream build

**Comment by domhanak** [comment]
@mariofusco I did fix the downstream build, please see referenced PRQ

**Pull request merged**